Updates the Install Minder Server documentation (#3880)
stacklok/minder: docs/docs/run_minder_server/run_the_server.md

---
title: Installing a Development version
sidebar_position: 10
---

# Installing a Development version 

This guide shows you how to run a Minder server locally. It is intended for users who would like to contribute to the Minder project. It is not intended for production use. 
This guide will walk you through how to: 
- Retrieve the latest source code
- Set up your development environment

> - Run the dependant applications

Reviewer: ```suggestion
- Run the dependent applications
```

cd minder
```

### Set up Development Environment
To set up your development environment, run:

```bash
make bootstrap
```
This will install the required tools for running different make targets, initialize required configuration files, as well as generate a token key passphrase.

### Build the application
Run the following to build minder and minder-server (binaries will be present at ./bin/)
```bash
make build
```

You may copy these into a location on your path, or run them directly from the `bin` directory.

### Configure the Repository Provider
You now need to create a Provider to enable Minder to inspect and manage your repository configuration. Currently only GitHub is supported as a Provider, so we'll do this using a GitHub App. This app will also provide Keycloak with an authentication source. Follow the steps in [Configuring a Provider](./config_provider.md) then return here to complete configuring the server. Be sure to save the Client ID and Client secret values, because you will need them again below. 

### Start the Minder server
At this point, you should have a GitHub provider configured and your `server-config.yaml` file updated. 

> Start `minder-server` along with its dependant services (`keycloak` and `postgres`) by running:

Reviewer: ```suggestion
Start `minder-server` along with its dependent services (`keycloak` and `postgres`) by running:
```

```bash
make run-docker
```

> As this command runs, you will see it build the Minder server container and then start the dependant containers.

Reviewer: ```suggestion
As this command runs, you will see it build the Minder server container and then start the dependent containers.
```

If you run
```bash
docker ps
```
you should see 4 new services running:
- keycloak
- minder
- openfga
- postgres

### Configure Keycloak

> Now that the Keycloak application is running, you need to configure it using the GitHub App you previous configured. 

Reviewer: ```suggestion
Now that the Keycloak application is running, you need to configure it using the GitHub App you previously configured. 
```

To enable GitHub login on Keycloak run the following command, using the `client_id` and `client_secret` you generated setting up the GitHub app: 

```bash
make KC_GITHUB_CLIENT_ID=<client_id> KC_GITHUB_CLIENT_SECRET=<client_secret> github-login
```

You should see it create a new instance and new mappers. You may see a resource not found message. This is safe to ignore. 

### Authenticate minder 
At this point, you should have the following:

> - A PostgreSQL database and Keycloak instance running in containers

Reviewer: ```suggestion
- A PostgreSQL database and Keycloak and OpenFGA instances running in containers
```
